invenio_github/api.py

```diff
@@ -623,6 +623,9 @@ def test_zipball(self):
             response.status_code == 200
         ), f"Could not retrieve archive from GitHub: {zipball_url}"
 
+        # Overwrite the original URL with the newly found more specific URL.
+        self.release_payload["zipball_url"] = zipball_url
```

Comment:
minor: I'm not sure if this is the best way to go about this, since we're effectively modifying the content of the webhook event we received from GitHub (temporarily in the fetched DB object, and I guess not actually persisting it in the DB).

Looking at this now, I'm somewhat confused/unhappy with how the whole function works. Shouldn't this just return the final/good Zipball URL? I would expect that this would be the result of `fetch_zipball_file` or `release_zipball_url`.

Reply:
Yes, it's a bit special:
* The `fetch_zipball_file` function downloads the ZIP file based on the `release_zipball_url` field.
* We [call `test_zipball`)](https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/services/github/release.py#L91) in order to modify what the `release_zipball_url` field will return before calling the `fetch_zipball_file` function.

I was going for a minimal fix, but I agree that it should be refactored.

Reply:
- [ ] Question: do we need the 2-step part of [first validating that the release files are fetchable](https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/services/github/release.py#L91) and [later initializing files](https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/services/github/release.py#L95-L100) ?